src/Stratis.FederatedPeg.Tests/TransactionBuilderTests.cs

```diff
@@ -128,43 +121,171 @@ public void BuildTransactionBuildsDeterministicTransactions()
+            Assert.Equal((uint256)2, transaction2.Inputs[1].PrevOut.Hash);
+            Assert.Equal((uint)0, transaction2.Inputs[1].PrevOut.N);
+
+            // Transaction outputs.
+            Assert.Equal(3, transaction2.Outputs.Count);
+
+            // Transaction output value - change.
+            Assert.Equal(new Money(9.98m, MoneyUnit.BTC), transaction2.Outputs[0].Value);
+            Assert.Equal(multiSigAddress.ScriptPubKey, transaction2.Outputs[0].ScriptPubKey);
+
+            // Transaction output value - recipient 2.
+            Assert.Equal(new Money(70m, MoneyUnit.BTC), transaction2.Outputs[1].Value);
+            Assert.Equal(recipient2[0].ScriptPubKey, transaction2.Outputs[1].ScriptPubKey);
+
+            // Transaction output value - op_return.
+            Assert.Equal(new Money(0m, MoneyUnit.BTC), transaction2.Outputs[2].Value);
+            Assert.Equal(opReturnData2, new OpReturnDataReader(this.loggerFactory, this.network).GetString(transaction2, out OpReturnDataType dummy2));
+
+            // Reserve UTXO's spent by transaction 2.
+            this.federationWalletManager.ProcessTransaction(transaction2);
+
+            // All the input UTXO's should be present in spending details of the multi-sig address.
+            Assert.True(this.SanityCheck(transaction2));
+
+        }
+
+        /// <summary>
+        /// Verifies that the transaction's inout UTXO's have been reserved by the wallet.
+        /// </summary>
+        /// <param name="transaction">The transaction to check.</param>
+        /// <returns><c>True</c> if all's well and <c>false</c> otherwise.</returns>
+        private bool SanityCheck(Transaction transaction)
+        {
+            // All the input UTXO's should be present in spending details of the multi-sig address.
+            foreach (TxIn input in transaction.Inputs)
+            {
+                TransactionData transactionData = this.wallet.MultiSigAddress.Transactions
+                    .Where(t => t.SpendingDetails != null && t.SpendingDetails.TransactionId == transaction.GetHash()
+                        && t.Id == input.PrevOut.Hash && t.Index == input.PrevOut.N).FirstOrDefault();
+
+                if (transactionData == null)
+                    return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Creates a directory for a test, based on the name of the class containing the test and the name of the test.
+        /// </summary>
+        /// <param name="caller">The calling object, from which we derive the namespace in which the test is contained.</param>
+        /// <param name="callingMethod">The name of the test being executed. A directory with the same name will be created.</param>
+        /// <returns>The path of the directory that was created.</returns>
+        public static string CreateTestDir(object caller, [System.Runtime.CompilerServices.CallerMemberName] string callingMethod = "")
```

Comment:
Maybe a new helper in Utils\ folder for those static methods ? no big deal for now